bindings/rust/generate/src/main.rs

```diff
@@ -199,7 +198,8 @@ impl FunctionCallbacks {
         let mut o = io::BufWriter::new(&mut tests);
 
         writeln!(o, "{}", COPYRIGHT)?;
-        for (feature, function) in functions.iter() {
+        let iter = functions.iter();
+        for (feature, function) in iter {
```

Comment:
why is this needed?

Reply:
Error:

```
> cargo clippy --manifest-path bindings/rust/Cargo.toml --all-targets -- -D warnings                        

    Checking generate v0.1.0 (/Volumes/workplace/s2n-tls/bindings/rust/generate)
    Checking s2n-tls v0.0.11 (/Volumes/workplace/s2n-tls/bindings/rust/s2n-tls)
error: temporary with significant `Drop` in `for` loop condition will live until the end of the `for` expression
   --> generate/src/main.rs:201:36
    |
201 |         for (feature, function) in functions.iter() {
    |                                    ^^^^^^^^^^^^^^^^
202 |             // don't generate tests for types
203 |             if types.contains(function) {
    |                ----- another value with significant `Drop` created here
...
208 |             if feature.is_some() && functions.contains(&(None, function.to_string())) {
    |                                     --------- another value with significant `Drop` created here
...
224 |         }
    |          - temporary lives until here
    |
    = note: `-D clippy::significant-drop-in-scrutinee` implied by `-D warnings`
    = note: this might lead to deadlocks or other unexpected behavior
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#significant_drop_in_scrutinee
```

Reply:
I'm hesitant to even fix this. This lint is unhelpful and doesn't make the code any more readable. In fact, I would argue it makes it less readable since it introduces an unnecessary variable in scope. From my reading, it looks like they're going to disable it by default in the next release: https://github.com/rust-lang/rust-clippy/issues/8987

Reply:
I agree it doesn't seem helpful, but I don't think it's bad enough to bother suppressing. One unnecessary variable seems fine.

Reply:
I think an extra `let` declaration is ok, especially if it keeps clippy happy. When clippy disables this lint, we can remove this change.

Reply:
> When clippy disables this lint, we can remove this change.

Not without a TODO and an issue or something we won't :) But I also don't think it's important enough to spend time reverting.

Reply:
Yeah not a blocker. Especially since this code isn't even consumed outside of publishing. I'm just kind of annoyed with clippy pushing new warnings by default that are not universally applicable.